integration-tests/src/scenario_utils.rs

```diff
@@ -0,0 +1,223 @@
+
+        contracts.insert(name.to_string(), compiled_contract);
+    }
+
+    contracts
+}
+
+/// Deploy contract
+pub async fn deploy_contract<M, T>(
+    wallet: &Arc<M>,
+    compiled: &CompiledContract,
+    args: T,
+) -> (u64, Contract<M>)
+where
+    M: Middleware,
+    T: Tokenize,
+{
+    let factory = ContractFactory::new(compiled.abi.clone(), compiled.bin.clone(), wallet.clone());
+    let (contract, receipt) = factory
+        .deploy(args)
+        .unwrap_or_else(|_| panic!("cannot construct contract creation tx: {:?}", compiled.name))
+        .confirmations(0usize)
+        .send_with_receipt()
+        .await
+        .unwrap_or_else(|_| panic!("cannot construct contract creation tx: {:?}", compiled.name));
+
+    (receipt.block_number.unwrap().as_u64(), contract)
+}
+
+/// A `sender` account sends ETH to multiple accounts
+pub async fn distribute_eth(
+    prov: &Provider<Http>,
+    sender: &WalletType,
+    wallets: &[WalletType],
+) -> u64 {
+    // Fund NUM_TXS wallets from wallet0
+    let mut pending_txs = Vec::new();
+    let mut nonce = prov
+        .get_transaction_count(sender.address(), None)
+        .await
+        .expect("cannot get transaction_count");
+
+    for wallet in wallets[0..NUM_TXS].iter() {
+        let tx = TransactionRequest::new()
+            .to(wallet.address())
+            .nonce(nonce)
+            .value(WEI_IN_ETHER * 2u8) // send 2 ETH
+            .from(sender.address());
+        pending_txs.push(
+            wallets[0]
+                .send_transaction(tx, None)
+                .await
+                .expect("cannot send tx"),
+        );
+        nonce = nonce.checked_add(U256::one()).unwrap();
+    }
+
+    wait_pending_txs(pending_txs).await
+}
+
+/// Wait pending transactions
+pub async fn wait_pending_txs(pending_txs: Vec<PendingTransaction<'_, Http>>) -> u64 {
+    let mut block_num = 0;
+    for tx in pending_txs.into_iter() {
+        let receipt = tx.await.expect("cannot confirm tx").unwrap();
+        if block_num == 0 {
+            block_num = receipt.block_number.unwrap().as_u64();
+        } else if block_num != receipt.block_number.unwrap().as_u64() {
+            panic!("The txs are not in a block")
+        }
+    }
+    block_num
+}
+
+/// TransferContext
+pub struct TransferContext {
+    /// Index of `from` wallet
+    pub from: usize,
+    /// Index of `to` wallet
+    pub to: usize,
+    /// Amount to transfer
+    pub amount: U256,
+    /// Expected result
+    pub expected: bool,
+}
+
+impl TransferContext {
+    /// init an instance
+    pub fn new(from_i: usize, to_i: usize, amount: U256, expected_result: bool) -> Self {
+        Self {
+            from: from_i,
+            to: to_i,
+            amount,
+            expected: expected_result,
+        }
+    }
+}
+
+/// Build and return erc20 transfer transaction
+/// TODO(dongchangYoo): Currently this function does not support sending multiple txs from an
+/// account in a
```

Comment:
```suggestion
/// account.
```
Sorry, I gave you an incomplete suggestion before.